src/Catalog/Monitoring/Validation/Test/PackageValidationTestRunner.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using NuGet.Services.Metadata.Catalog.Monitoring.Validation.Result;
+using NuGet.Packaging.Core;
+using NuGet.Services.Metadata.Catalog.Persistence;
+using System.Threading;
+using Microsoft.Extensions.Logging;
+using NuGet.Services.Metadata.Catalog.Helpers;
+using System;
+using Newtonsoft.Json.Linq;
+using NuGet.Versioning;
+
+namespace NuGet.Services.Metadata.Catalog.Monitoring.Validation.Test
+{
+    /// <summary>
+    /// Stores a set of <see cref="EndpointValidationTestRunner"/>s and runs them.
+    /// </summary>
+    public class PackageValidationTestRunner
+    {
+        private readonly IEnumerable<EndpointValidationTestRunner> _endpointValidationTestRunners;
+
+        public PackageValidationTestRunner(IEnumerable<EndpointValidationTestRunner> endpointValidationTestRunners)
+        {
+            _endpointValidationTestRunners = endpointValidationTestRunners;
+        }
+
+        /// <summary>
+        /// Runs <see cref="IValidationTestRunner"/>s from the <see cref="EndpointValidationTestRunner"/>s against a package.
+        /// </summary>
+        /// <returns>A <see cref="PackageValidationTestResult"/> generated from the results of the <see cref="IValidationTestRunner"/>s.</returns>
+        public async Task<PackageValidationTestResult> Run(string packageId, string packageVersion, IList<JObject> catalogEntriesJson, Storage auditingStorage, CollectorHttpClient client, CancellationToken cancellationToken, ILogger logger)
```

Comment:
I pass the version as a string to this function so there is no chance that it will throw outside of the validation.